src/Microsoft.Dnx.Runtime/Compilation/ICompilationSession.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Dnx.Runtime.Compilation
+{
+    public interface ICompilationSession : IDisposable
+    {
+        event Action<string> OnInputFileChanged;
+
+        Assembly CompileAndLoadProject(Project project, CompilationTarget target, IAssemblyLoadContext loadContext);
```

Comment:
The idea here is that all the compilation engine is needed for, in the **runtime**, is to actually compile and load the project's assembly.